### PR TITLE
Pull development into stable

### PR DESCRIPTION
**Pull request by coltenkrauter**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -183,14 +269,14 @@
...
+              "devServerTarget": "angular-ssr-example:serve:prod"
             },
-            "certification": {
-              "devServerTarget": "angular-ssr-example:serve:certification"
+            "cert": {
+              "devServerTarget": "angular-ssr-example:serve:cert"
             },
-            "development": {
-              "devServerTarget": "angular-ssr-example:serve:development"
+            "dev": {
```
Comment (calebkrauter) [inline angular.json:278]: I like your clean naming style with use of "prod", "dev", "cert" and the like.

File: src/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  imports: [
+    BrowserModule.withServerTransition({ appId: "serverApp" }),
+    GaugeModule.forRoot(),
+    CoreModule,
+    AppRoutingModule,
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
```
Comment (calebkrauter) [inline src/app.module.ts:23]: Do you consolidate your imports to just a few classes and then export it for use in other classes for code-cleanliness? I realize that was possible! Very cool.

File: src/environments/environment.prod.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const environment = {
```
Comment (calebkrauter) [inline src/environments/environment.prod.ts:1]: What is the function and purpose of the environment constant? src/environments/environment.dev-ssr.ts, src/environments/environment.dev.ts, src/environments/environment.prod-ssr.ts and src/environments/environment.prod.ts 

Why are there multiple that look very similar?

**Review by calebkrauter — COMMENTED** [review]
Looks interesting. It appears that you are automating something. I am not very familiar with Docker, I'll do some research on it.

File: src/app.module.ts (hunk 2 above)
Comment (coltenkrauter) [reply src/app.module.ts:23]: I don't remember tbh... iirc, this was a requirement of the Angular library.

File: src/environments/environment.prod.ts (hunk 3 above)
Comment (coltenkrauter) [reply src/environments/environment.prod.ts:1]: These serve as configs that will be selected based on the given environment to configure the app.

**Comment by coltenkrauter** [comment]
> Looks interesting. It appears that you are automating something. I am not very familiar with Docker, I'll do some research on it.

Are you referring to the entire PR? Ya, at one point in time, it would deploy environments automatically with the help of https://drone.io.

**Pull request merged**